p2p/src/authenticated/discovery/actors/router/actor.rs

```diff
@@ -121,7 +129,13 @@ impl<E: Spawner + Metrics, P: PublicKey> Actor<E, P> {
                             // Send to all connected peers
                             for (recipient, messenger) in self.connections.iter_mut() {
                                 if messenger
-                                    .content(channel, message.clone(), priority)
+                                    .send(
+                                        Data {
+                                            channel,
+                                            message: message.clone(),
```

Comment:
nit: not in your code, but is the clone on line 121 unnecessary? And if so, I would think that clippy should catch it, so not sure what's going on there

Reply:
We need the clone because we're calling `send` in a loop, and each function call consumes the message.

That said, rust is happy to let you `clone()` all you want. After all, you are _explicitly_ saying that you want to copy something. However, it won't _implicitly_ copy something (i.e. without `clone()`) unless you make that something `Copy` to signify that it's cheap enough that it's not worth worrying about.